### PR TITLE
execute: don't honour PrivateNetwork() if we lack CAP_NET_ADMIN

### PR DESCRIPTION
**Pull request by poettering**
Somehow the Linux kernel allows us to allocate a network namespace if we possess CAP_SYS_ADMIN but doesn't allow us to configure it, unless we also have CAP_NET_ADMIN.

Taking that into consideration let's avoid allocating a network namespace we cannot even configure "lo" in.

This is common case if nspawn is invoked without userns and without netns, because in that case it will have CAP_SYS_ADMIN but no CAP_NET_ADMIN.

This also takes down a notch the messages about the automatic downgrading. These have been LOG_WARNING so far, and I downgraded them to LOG_NOTICE, since in an environment where CAP_NET_ADMIN is not available this is really not something to be concerned about, but still noticable. With that it's still more priorized than regular LOG_INFO.

Fixes: #27292

### CODE REVIEW

File: src/core/execute.c [hunk 1]
```diff
@@ -5412,12 +5412,16 @@ static int exec_child(
...
+                 * new network namespace. And if we don't have that, then we could only create a network
+                 * namespace without the ability to set up "lo". Hence gracefully skip things then. */
+                if (ns_type_supported(NAMESPACE_NET) && have_effective_cap(CAP_NET_ADMIN) != 0) {
                         r = setup_shareable_ns(runtime->shared->netns_storage_socket, CLONE_NEWNET);
                         if (r < 0) {
                                 if (ERRNO_IS_PRIVILEGE(r))
-                                        log_unit_warning_errno(unit, r,
-                                                               "PrivateNetwork=yes is configured, but network namespace setup failed, ignoring: %m");
+                                        log_unit_notice_errno(unit, r,
+                                                               "PrivateNetwork=yes is configured, but network namespace setup not permitted, proceeding wthout: %m");
```
Comment (YHNdnzj) [inline src/core/execute.c]:
```suggestion
                                                               "PrivateNetwork=yes is configured, but network namespace setup not permitted, proceeding without: %m");
```

File: src/core/execute.c [hunk 2]
```diff
@@ -5412,12 +5412,16 @@ static int exec_child(
 
         if (exec_needs_network_namespace(context) && runtime && runtime->shared && runtime->shared->netns_storage_socket[0] >= 0) {
 
-                if (ns_type_supported(NAMESPACE_NET)) {
+                /* Try to enable network namespacing if network namespacing is available and we have
+                 * CAP_NET_ADMIN. We need CAP_NET_ADMIN to be able to configure the loopback device in the
+                 * new network namespace. And if we don't have that, then we could only create a network
+                 * namespace without the ability to set up "lo". Hence gracefully skip things then. */
+                if (ns_type_supported(NAMESPACE_NET) && have_effective_cap(CAP_NET_ADMIN) != 0) {
```
Comment (keszybz) [inline src/core/execute.c]: Not that it matters greatly, but in other places we treat failure as negative. So maybe say `have_effective_cap(CAP_NET_ADMIN) > 0` here too?

File: src/core/execute.c [hunk 3]
```diff
@@ -5428,7 +5432,7 @@ static int exec_child(
                         return log_unit_error_errno(unit, SYNTHETIC_ERRNO(EOPNOTSUPP),
                                                     "NetworkNamespacePath= is not supported, refusing.");
                 } else
-                        log_unit_warning(unit, "PrivateNetwork=yes is configured, but the kernel does not support network namespaces, ignoring.");
+                        log_unit_notice(unit, "PrivateNetwork=yes is configured, but the kernel or available privileges do not support or permit network namespaces, proceeding without.");
```
Comment (keszybz) [inline src/core/execute.c]: ", but the kernel does not support or we lack privileges for network namespace, proceeding without" ?

**Review by keszybz — COMMENTED** [review]
Thanks for looking into this.

**Review by bluca — APPROVED** [review]

**Pull request merged**